docs/development/migrations.md

```diff
@@ -0,0 +1,22 @@
+Migrations
+==========
+Kubernikus incorporates a simple migration system to update the internal representation of existing Klusters.
+This can be used to update the CRD of a Kluster or update any other accompanying kubernetes resource in the control plane (e.g. Kluster Secret, Ingress, etc).
+
+To create a migration a [Migration](https://github.com/sapcc/kubernikus/blob/master/pkg/migration/migration.go) function needs to be written and [registered](https://github.com/sapcc/kubernikus/blob/master/pkg/migration/register.go).
+
+```
+type Migration func(klusterRaw []byte, kluster *v1.Kluster, client kubernetes.Interface) (err error)
+```
+* `klusterRaw` contains the current raw JSON data of the Kluster, it can be used to access old data not exposed by the `v1.Kluster` type anymore.
```

Comment:
I have a vague recollection that we discussed about this, but why is `klusterRaw` a `json` blob? Wouldn't it be easier to access if it were a deep-copy of `kluster`?

Reply:
The problem is that the migration runs in the new codebase with the new `CRD` model. That means if you removed a field (or renamed, changed the structure) you loose the old data. So loading it into a `Kluster` is not an option.